warehouse/scripts/dbt_artifacts/sources.py

```diff
@@ -0,0 +1,103 @@
+# generated by datamodel-codegen:
+#   filename:  https://schemas.getdbt.com/dbt/sources/v3.json
+#   timestamp: 2023-05-15T18:21:37+00:00
+
+from __future__ import annotations
+
+from datetime import datetime
+from enum import Enum
+from typing import Any, Dict, List, Optional, Union
+
+from pydantic import BaseModel, Extra, Field
+
+
+class FreshnessMetadata(BaseModel):
+    class Config:
+        extra = Extra.allow
+
+    dbt_schema_version: Optional[str] = "https://schemas.getdbt.com/dbt/sources/v3.json"
+    dbt_version: Optional[str] = "1.0.0b2"
+    generated_at: Optional[datetime] = "2021-11-02T20:18:06.796684Z"
+    invocation_id: Optional[str] = None
+    env: Optional[Dict[str, str]] = {}
+
+
+class Status(Enum):
+    runtime_error = "runtime error"
+
+
+class SourceFreshnessRuntimeError(BaseModel):
+    class Config:
+        extra = Extra.allow
+
+    unique_id: str
+    error: Optional[Union[str, int]] = None
+    status: Status
+
+
+class Status1(Enum):
+    pass_ = "pass"
+    warn = "warn"
+    error = "error"
+    runtime_error = "runtime error"
+
+
+class PeriodEnum(Enum):
```

Comment:
I assume it would be more trouble than it's worth to try to dedupe but it's a bit weird to have identical enums defined over and over again

Reply:
yes I've been meaning to open an upstream issue asking why they were defined like this in the json schema